### PR TITLE
Shrink db size

### PR DESCRIPTION
**Pull request by simonjiao**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error handling for unclean database states in blockchain initialization.

- **Refactor**
  - Removed outdated methods and streamlined data retrieval processes in the consensus header and ghostdag databases.
  - Improved configuration handling for database settings.

### CODE REVIEW

File: flexidag/src/consensusdb/consensus_ghostdag.rs [hunk 1]
```diff
@@ -232,15 +227,6 @@ impl DbGhostdagStore {
         }
         self.access
             .write(BatchDbWriter::new(batch), hash, data.clone())?;
-        self.compact_access.write(
-            BatchDbWriter::new(batch),
-            hash,
-            CompactGhostdagData {
```
Comment (nkysg) [inline flexidag/src/consensusdb/consensus_ghostdag.rs:238]: We don't need CompactGhostdagData? If we can remove struct CompactGhostdagData

**Review by nkysg — APPROVED** [review]

File: flexidag/src/consensusdb/consensus_ghostdag.rs (hunk 1 above)
Comment (simonjiao) [reply flexidag/src/consensusdb/consensus_ghostdag.rs:238]: Currently, we can remove some CF from rocksdb, such as DagHeader and CompactGhostDag. Actually they're not used or partially used in dag. But the db size seems not become smaller for an overnight test. lol. Need more investigation.

**Comment by nkysg** [comment]
@coderabbitai full review

**Comment by simonjiao** [comment]
@coderabbitai review

**Pull request closed without merging**